src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -146,16 +139,47 @@ public void UnregisterProjectConfiguration(Guid projectGuid, ConfiguredProject p
 
                 _projectFaultHandlerService.Forget(displayMissingComponentsTask, project: project.UnconfiguredProject, ProjectFaultSeverity.Recoverable);
             }
+
+            void RemoveConfiguration(Guid projectGuid, ConfiguredProject project)
+            {
+                IConcurrentHashSet<ProjectConfiguration>? projectConfigurationSet;
+
+                if (projectGuid == Guid.Empty)
+                {
+                    ProjectPathToProjectConfigurationsMap.TryGetValue(project.UnconfiguredProject.FullPath, out projectConfigurationSet);
+                }
+                else
+                {
+                    _projectGuidToProjectConfigurationsMap.TryGetValue(projectGuid, out projectConfigurationSet);
+                }
+                
+                projectConfigurationSet?.Remove(project.ProjectConfiguration);
+            }
         }
 
         private bool ShouldDisplayMissingComponentsPrompt()
         {
             lock (_displayPromptLock)
             {
-                return (_projectGuidToWorkloadDescriptorsMap.Count > 0 || _projectGuidToRuntimeDescriptorMap.Count > 0)
-                    && _projectGuidToProjectConfigurationsMap.Values.All(projectConfigurationSet => projectConfigurationSet?.Count == 0)
-                    && (_projectPathToProjectConfigurationsMap == null ||
-                        _projectPathToProjectConfigurationsMap.Values.All(projectConfigurationSet => projectConfigurationSet?.Count == 0));
+                // Projects that subscribe to this service will registers all their configurations and after that
+                // each project configuration can start registering missing workload at different point in time.
+                // We want to display the prompt after ALL the registered project already registered their missing components
+                // and at least there is one component to install.
+                return AreMissingComponentsToInstall()
+                    && AllProjectsConfigurationsRegisteredTheirMissingComponents();
+            }
+
+            bool AreMissingComponentsToInstall()
+            {
+                // Projects can register zero or more missing components.
+                return _projectGuidToWorkloadDescriptorsMap.Count > 0 || _projectGuidToRuntimeDescriptorMap.Count > 0;
+            }
+
+            bool AllProjectsConfigurationsRegisteredTheirMissingComponents()
+            {
+                // When a project configuration registers its missing components, the configuration gets removed, but we keep the list of components.
+                return _projectGuidToProjectConfigurationsMap.Values.All(projectConfigurationSet => projectConfigurationSet.Count == 0)
+                    && _projectPathToProjectConfigurationsMap?.Values.All(projectConfigurationSet => projectConfigurationSet.Count == 0) is null or true;
```

Comment:
If you continue working in this code, I wonder if you could make this non-nullable. While it isn't assigned for every solution, it's in global scope so would only be a single allocation for the whole IDE process, which isn't worth worrying about. Making it non-null would allow removing the `ProjectPathToProjectConfigurationsMap` property too, and using the field directly.